src/main/java/frc/trigon/robot/poseestimation/poseestimator/PoseEstimator.java

```diff
@@ -196,22 +253,28 @@ private void sortCamerasByLatestResultTimestamp(AprilTagCamera[] aprilTagCameras
     }
 
     /**
-     * Updates the estimated pose from each camera, each at the timestamp of their latest result.
+     * Sets the estimated pose from a pose source at the given timestamp.
+     *
+     * @param observationPose the estimated robot pose
+     * @param timestamp       the timestamp of the observation
      */
-    private void updateEstimatedPoseFromVision(AprilTagCamera[] aprilTagCameras) {
-        for (AprilTagCamera aprilTagCamera : aprilTagCameras) {
-            addVisionObservation(
-                    aprilTagCamera.getEstimatedRobotPose(),
-                    aprilTagCamera.calculateStandardDeviations(),
-                    aprilTagCamera.getLatestResultTimestampSeconds()
-            );
-        }
+    private void addPoseSourceObservation(Pose2d observationPose, double timestamp, StandardDeviations standardDeviations) {
+        final Pose2d estimatedPoseAtObservationTime = getPoseAtTimestamp(timestamp);
+        if (estimatedPoseAtObservationTime == null)
+            return;
+
+        this.estimatedPose = calculateEstimatedPoseWithAmbiguityCompensation(estimatedPoseAtObservationTime, observationPose, standardDeviations);
     }
 
-    private void addVisionObservation(Pose2d observationPose, StandardDeviations standardDeviations, double timestamp) {
-        final Pose2d estimatedPoseAtObservationTime = getPoseAtTimestamp(timestamp);
-        if (estimatedPoseAtObservationTime != null)
-            this.estimatedPose = calculateEstimatedPoseWithAmbiguityCompensation(estimatedPoseAtObservationTime, observationPose, standardDeviations);
+    private boolean isPoseOutOfHeldPosesRange(double timestamp) {
```

Comment:
The name is misleading, this isn't checking anything about a pose, but about a timestamp.